pennylane/transforms/transpile.py

```diff
@@ -14,9 +16,51 @@
 from pennylane.tape import QuantumTape
 
 
+def state_transposition(results, mps, new_wire_order, original_wire_order):
+    """Transpose the order of any state return.
+
+    Args:
+        results (ResultBatch): the result of executing a batch of length 1
+
+    Keyword Args:
+        mps (List[MeasurementProcess]): A list of measurements processes. At least one is a ``StateMP``
+        new_wire_order (Sequence[Any]): the wire order after transpile has been called
+        original_wire_order (.Wires): the devices wire order
+
+    Returns:
+        Result: The result object with state dimensions transposed.
+
+    """
+    if len(mps) == 1:
+        temp_mp = qml.measurements.StateMP(wires=original_wire_order)
+        return temp_mp.process_state(results[0], wire_order=qml.wires.Wires(new_wire_order))
+    new_results = list(results[0])
+    for i, mp in enumerate(mps):
+        if isinstance(mp, qml.measurements.StateMP):
+            temp_mp = qml.measurements.StateMP(wires=original_wire_order)
+            new_res = temp_mp.process_state(
+                new_results[i], wire_order=qml.wires.Wires(new_wire_order)
+            )
+            new_results[i] = new_res
+    return tuple(new_results)
+
+
+def _process_measurements(expanded_tape, device_wires, is_default_mixed):
+    measurements = expanded_tape.measurements.copy()
+    if device_wires:
+        for i, m in enumerate(measurements):
+            if isinstance(m, qml.measurements.StateMP):
```

Comment:
why not add wires to stateMP if not default_mixed?

Reply:
Because then other devices will interpret it as a density matrix instead of a state measurement.

Reply:
```
>>> dev = qml.device('lightning.qubit', wires=1)
>>> tape = qml.tape.QuantumScript([],[qml.measurements.StateMP(wires=qml.wires.Wires(0))])
>>> dev.execute(tape)
array([[1.+0.j, 0.+0.j],
       [0.+0.j, 0.+0.j]])
```